### DISCUSSION TITLE
Milvus migration from Pulsar to Kafka

### DISCUSSION THREAD

**Discussion by dutysloth** [milvus-io/milvus#31971] [Q&A and General discussion]
Hi,
I have a working Milvus cluster deployment that is currently using Pulsar (internal, not external) as the log broker.
I would like to start using an external Kafka cluster as my log broker.
Is there any recommended procedure for migration?

Thanks in advance

**Comment by xiaofan-luan** [comment]
> Hi, I have a working Milvus cluster deployment that is currently using Pulsar (internal, not external) as the log broker. I would like to start using an external Kafka cluster as my log broker. Is there any recommended procedure for migration?
> 
> Thanks in advance

@LoveEachDay this is actaully a interesting topic. I'm assuming we can not do this due to the checkpoint format. Did we do something similar in house?
